system-addon/content-src/components/Search/Search.jsx

```diff
@@ -19,20 +20,25 @@ class Search extends React.Component {
     }
   }
   onClick(event) {
-    this.controller.search(event);
+    window.gContentSearchController.search(event);
+  }
+  componentWillUnmount() {
+    delete window.gContentSearchController;
   }
   onInputMount(input) {
     if (input) {
-      // The first "newtab" parameter here is called the "healthReportKey" and needs
-      // to be "newtab" so that BrowserUsageTelemetry.jsm knows to handle events with
-      // this name, and can add the appropriate telemetry probes for search. Without the
-      // correct name, certain tests like browser_UsageTelemetry_content.js will fail (See
-      // github ticket #2348 for more details)
-      this.controller = new ContentSearchUIController(input, input.parentNode,
-        "newtab", "newtab");
+      // The first parameter here is called the "healthReportKey" and needs
+      // to be "newtab" or "abouthome" so that BrowserUsageTelemetry.jsm knows to
+      // handle events with this name, and can add the appropriate telemetry
+      // probes for search. Without the correct name, certain tests like
+      // browser_UsageTelemetry_content.js will fail (See github ticket #2348 for more details)
+      const healthReportKey = IS_NEWTAB ? "newtab" : "abouthome";
+      const source = IS_NEWTAB ? "newtab" : "homepage";
```

Comment:
Let's name this `searchSource` and also have a comment describing that it's so search engine plugins can correctly attribute referrals especially.